k9mail/src/main/java/com/fsck/k9/activity/compose/AttachmentPresenter.java

```diff
@@ -175,6 +185,27 @@ public void processMessageToForward(MessageViewInfo messageViewInfo) {
         }
     }
 
+    public void processMessageToForwardAsAttachment(MessageViewInfo messageViewInfo) throws IOException, MessagingException {
+        boolean isMissingParts = !allPartsAvailable(messageViewInfo, false);
+        if (isMissingParts) {
+            attachmentMvpView.showMissingAttachmentsPartialMessageForwardWarning();
+        } else {
+            File tempFile = File.createTempFile("pre", ".tmp");
+            tempFile.deleteOnExit();
```

Comment:
We should pick one of the strategies outlined here: https://developer.android.com/reference/java/io/File.html#deleteOnExit()

I think the Unix trick is too low level. The list idea may work, but the finally strategy is neatest if we can refactor the code accordingly.

This probably wants an audit issue?